core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -35,7 +35,7 @@
  */
 public class RingBufferLogHandler extends Handler {
 
-    private static final int DEFAULT_RING_BUFFER_SIZE = Integer.getInteger(RingBufferLogHandler.class.getName() + ".defaultSize", 256);
+    public static final int DEFAULT_RING_BUFFER_SIZE = 256;
```

Comment:
Why did you change this `= Integer.getInteger(RingBufferLogHandler.class.getName() + ".defaultSize", 256);` to just ` = 256`? 
With the former way, we can change the buffer size with a System property, chich could be interesting.

Reply:
```RingBufferLogHandler.class.getName() + ".defaultSize"``` was flawed for example because the .defaultSize property does not exist and the function attributed by default 256. https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#getInteger(java.lang.String,%20int)
That function is an reflexive and unnecessary .
From the ticket it is said "Currently, as an intermediate shim, I'm using reflection. In addition to being dirty, this is likely to become even more an issue to support Java 9 when using such hacks is being made harder and harder." I agree with that statement.
Attributing an integer is better in my opinion.

Reply:
If I keep that previous function I need to add a int defaultSize field that is itself 256, that serves no purpose.

Reply:
I'm pretty sure @batmat was refferring to [these lines](https://github.com/jenkinsci/evergreen-plugin/commit/5f42dc4e3cca1981dc7c5636b4139731c0d16ec4#diff-4a65f41c4b96b07b21810d8bd8a51161R82-R83) when he said _being dirty, this is likely to become even more an issue to support Java 9_ . There is nothing wrong with using a system property to get the default value, it has nothing to do with reflection either. 

The reflective call is https://github.com/jenkinsci/evergreen-plugin/commit/5f42dc4e3cca1981dc7c5636b4139731c0d16ec4#diff-4a65f41c4b96b07b21810d8bd8a51161R84 and the issue with Java9 and higher is https://github.com/jenkinsci/evergreen-plugin/commit/5f42dc4e3cca1981dc7c5636b4139731c0d16ec4#diff-4a65f41c4b96b07b21810d8bd8a51161R83

Please consider keeping the former behavior, but `public`.